src/ast/passes/semantic_analyser.cpp

```diff
@@ -1104,7 +1104,12 @@ void SemanticAnalyser::visit(Call &call)
     call.type = CreateNone();
   } else if (call.func == "exit") {
     check_assignment(call, false, false, false);
-    (void)check_nargs(call, 0);
+
+    if (!check_varargs(call, 0, 1))
+      return;
+
+    if (call.vargs.size() == 1)
+      check_arg(call, Type::integer, 0);
```

Comment:
I need to read through this a bit more but maybe we need to limit the size of the exit code to an int8
https://pubs.opengroup.org/onlinepubs/9699919799/functions/_exit.html

Reply:
good catch! actually we're using int8 in other parts of the [code](https://github.com/bpftrace/bpftrace/blob/c67683752655258a0b730840b95dcdf84ef2f8ec/src/ast/passes/codegen_llvm.cpp#L1174). will change it.

Reply:
Did this get missed in an update?

I think for a int literal we can probably check the range and for a variable we can insert a cast to int8 perhaps.

Reply:
I guess we don't need to check the range as it says
> only the least significant 8 bits (that is, status & 0377) shall be available from wait() and waitpid()

Reply:
Ok, but you'll still need a cast somewhere (this is why you're getting errors on LLVM 13/14. You can add the conditional codegen cast like I did here:
https://github.com/bpftrace/bpftrace/blob/master/src/ast/passes/codegen_llvm.cpp#L782